source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -10,6 +10,46 @@ namespace Extensions {
 namespace TransportSockets {
 namespace Tls {
 
+namespace {
+
+enum class CertName { Issuer, Subject };
+
+/**
+ * Retrieves a name from a certificate and formats it as an RFC 2253 name.
+ * @param cert the certificate.
+ * @param desired_name the desired name (Issuer or Subject) to retrieve from the certificate.
+ * @return std::string returns the desired name formatted as an RFC 2253 name.
+ */
+std::string getRFC2253NameFromCertificate(X509& cert, CertName desired_name) {
+  bssl::UniquePtr<BIO> buf(BIO_new(BIO_s_mem()));
+  RELEASE_ASSERT(buf != nullptr, "");
+
+  X509_NAME* name = nullptr;
+  switch (desired_name) {
+  case CertName::Issuer:
+    name = X509_get_issuer_name(&cert);
+    break;
+  case CertName::Subject:
+    name = X509_get_subject_name(&cert);
+    break;
+  }
```

Comment:
Usually we'll add a default case with a `NOT_REACHED_GCOVR_EXCL_LINE` statement to help detect missing cases.

Reply:
Fixed, thank you!

Reply:
This default shouldn't be needed at all since you cover all cases. Can you remove?

Reply:
@mattklein123 Sure -- I've removed it in the latest commit.